flowmachine/flowmachine/features/location/majority_location.py

```diff
@@ -0,0 +1,35 @@
+from typing import List
+
+from flowmachine.core import Query
+from flowmachine.core.mixins import GeoDataMixin
+from flowmachine.features.utilities.subscriber_locations import BaseLocation
+
+
+class MajorityLocation(BaseLocation, Query):
+    def __init__(self, subscriber_subset: Query, weight_column):
```

Comment:
I'd suggest something like 'subscriber_location_weights' instead of 'subscriber_subset'